master/getting-started/openstack/upgrade/index.md

```diff
@@ -0,0 +1,67 @@
+---
+title: Upgrading Calico for OpenStack
+canonical_url: 'https://docs.projectcalico.org/master/getting-started/openstack/upgrade/'
+---
+## Upgrade procedure (OpenStack)
+These documents detail the procedure for upgrading {{site.prodname}} in an OpenStack system. 
+If a release does not upgrade a particular component, you should skip the steps for that component.
+
+> **Important**: You will
+> be unable to issue API requests to your OpenStack system during the
+> procedure. Please plan your upgrade window accordingly, and see the
+> [Service Impact](#service-impact) section for more details.
+{: .alert .alert-danger}
+
+## Service impact
+
+During the upgrade, **all VMs will continue to function normally**:
+there should be no impact on the data plane. However, control plane
+traffic may fail at different points throughout the upgrade.
+
+Generally, users should be prevented from creating or updating virtual
+machines during this procedure, as these actions will fail. VM deletion
+*may* succeed, but will likely be delayed until the end of the upgrade.
+
+For this reason, we highly recommend planning a maintenance window for
+the upgrade. During this window, you should disable all user API access
+to your OpenStack deployment.
+
+
+## Before you begin
+
+- You must first [upgrade](/v2.6/getting-started/openstack/upgrade) 
+  to {{site.prodname}} [v2.6.x](https://github.com/projectcalico/calico/releases) 
+  before you can upgrade to {{site.prodname}} 
+  {{site.data.versions[page.version].first.title}}. 
+
+## Upgrade steps
+
+If you use calicoctl to add Calico policies in addition to those that are derived automatically (by our Neutron driver) from OpenStack security groups - or more generally to configure any Calico data other than the data that is generated by the Calico Neutron driver - you will need to follow the steps marked OPTIONAL below. Otherwise you can skip those OPTIONAL steps.
+
+1. Set up an etcdv3 cluster, if you don't already have etcdv3 capability, either by [upgrading the etcd software of your existing etcdv2 cluster](https://coreos.com/etcd/docs/latest/upgrades/upgrade_3_0.html), or by [installing an etcdv3 cluster on new servers](https://coreos.com/etcd/docs/latest/op-guide/clustering.html).
+  
+   > **Tip**: etcd v3.x still supports the etcd v2 API, as used by {{site.prodname}} v2.6.x.
```

Comment:
It might be nice if we could make this note consistent with the body text in terms of how we refer to etcd and its versions. In the body text, we don't have `etcd v3.x` or `etcd v3` or `etcd v2`, it is `etcdv3` and `etcdv2` there. This is not a major point, but just something if possible to clean up.